Dockerfile

```diff
@@ -0,0 +1,9 @@
+FROM python:3.7.7
+COPY . /StopSpot_Data_Pipeline
+RUN python -m pip install \
+        pipenv \
+        parse \
+        realpython-reader \
+        pandas \
+        sqlalchemy \
+        psycopg2

```

Comment:
I probably missed this but why is it we can't do pipenv install?

Reply:
This I don't have a good answer for but I believe Nelson had mentioned something at one point. I'll try playing around with this a bit.

Reply:
So a little update, I don't have a for certain answer but I think I am starting to see a potential issue that would arise. So when passing these docker commands through the docker run command it is passing the commands through a non interactive session. Doing a pipenv install works fine and dandy but when trying to do a pipenv shell it complains with the following stack trace.

```
✔ Successfully created virtual environment! 
Virtualenv location: /root/.local/share/virtualenvs/StopSpot_Data_Pipeline-flWWEnci
Installing dependencies from Pipfile.lock (821d8c)…
To activate this project's virtualenv, run pipenv shell.
Alternatively, run a command inside the virtualenv with pipenv run.
Launching subshell in virtual environment…
Traceback (most recent call last):
  File "/usr/local/bin/pipenv", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pipenv/vendor/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pipenv/vendor/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/pipenv/cli/command.py", line 390, in shell
    pypi_mirror=state.pypi_mirror,
  File "/usr/local/lib/python3.7/site-packages/pipenv/core.py", line 2184, in do_shell
    shell.fork_compat(*fork_args)
  File "/usr/local/lib/python3.7/site-packages/pipenv/shells.py", line 121, in fork_compat
    c.interact(escape_character=None)
  File "/usr/local/lib/python3.7/site-packages/pipenv/vendor/pexpect/pty_spawn.py", line 761, in interact
    mode = tty.tcgetattr(self.STDIN_FILENO)
termios.error: (25, 'Inappropriate ioctl for device')
```
Looking into that error a little bit the high level issue is that it has to do with a non interactive environment and therefore user input is nullified. So by installing the dependencies it would allow us to run the application while still using a non interactive environment. Additonally, I did try using pipenv run but it complained with

`the command main.py could not be found within PATH or Pipfile's [scripts]`

Perhaps you know a work around to the above line?

Reply:
I do not know very much about docker or how it interacts with Pipenv, so if you and Nelson have been having issues, I am probably not going to have much luck. I think that installing the packages all manually in the dockerfile wouldn't be that bad of an alternative, but it seems undesirable to have to adjust this file manually should packages be changed later. Maybe if there was a way to check `Pipefile` for the packages it would load and load them dynamically, that would be a bit more flexible. But I also am not super aware of how this would all shake down so I'll leave this to your judgement.